tools/harness-thci/OpenThread_BR.py

```diff
@@ -303,6 +304,9 @@ def _disconnect(self):
             self.__handle = None
 
     def _deviceBeforeReset(self):
+        if self.isPowerDown:
+            self.log('Powering up the device')
```

Comment:
why do we power up the device before reset?

Reply:
it is for the case, where we called powerDown - in that case, we stop otbr-agent service and set isPowerDown flag to True. Then we call reset() and wait for the thread state returning disabled/detached - since the service is down (and the flag is True) we will get an error each time we call state command and the reset function fails - this is actually the main goal of this PR to fix this issue.

Reply:
SGTM